woocommerce/woocommerce
# E2e/gutenberg latest nightly 0

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This PR replaces "WordPress SEO" and "Contact Form 7" with "Gutenberg" and "Gutenberg - Nightly" in the matrix of plugins that our daily and release smoke tests use.

Other minor changes include:

- Use the PNPM script `pnpm test:e2e-pw` instead of the longer `pnpm exec playwright...` command as shown [here](https://github.com/woocommerce/woocommerce/pull/38287/files#diff-3b2ff7d93606602bcae1ce86708096f7755e592f68aff14ade8d6656027b81fcL227-R229).
- Add `test.step()`'s in `upload-plugin` spec.
- Add 2 assertions in the `upload-plugin` spec to make sure the shop and the WooCommerce homepage loads successfully after installing the plugin



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

#### Test the `upload-plugin` spec locally

1. Run the usual setup commands to launch the E2E environment locally.
1. Run the `upload-plugin` spec with `Gutenberg` plugin by running these commands in the `plugins/woocommerce` directory, and make sure the test passed:
   ```
   export PLUGIN_NAME="Gutenberg"
   export PLUGIN_REPOSITORY="WordPress/gutenberg"
   pnpm test:e2e-pw upload-plugin.spec.js
   ```
1. Run the `upload-plugin` spec with `Gutenberg - Nightly` plugin by running these commands in the `plugins/woocommerce` directory, and make sure the test passed:
   ```
   export PLUGIN_NAME="Gutenberg - Nightly"
   export PLUGIN_REPOSITORY="bph/gutenberg"
   pnpm test:e2e-pw upload-plugin.spec.js
   ```

#### Test the "Smoke test daily" workflow

1. Go to the [Smoke test daily](https://github.com/woocommerce/woocommerce/actions/workflows/smoke-test-daily.yml) page.
1. Select this branch `e2e/gutenberg-latest-nightly-0` and start the workflow.
1. Wait for all the jobs to finish.
1. Go to the [WooCommerce Test Reports Dashboard > Daily smoke tests](https://woocommerce.github.io/woocommerce-test-reports/daily/) page.
1. Verify that clicking the `E2E` links for `Gutenberg` and `Gutenberg - Nightly` will take you to the correct Allure reports
   - Make sure that the plugin name, time, and date shown in the Allure report are correct.
     <img width="389" alt="xRvFGUMh4c" src="https://github.com/woocommerce/woocommerce/assets/4509348/69fbec69-9c8a-4f9a-ac3e-b182d0914b39">
     <img width="374" alt="5NoFlm2wEO" src="https://github.com/woocommerce/woocommerce/assets/4509348/cf4f72e7-0447-465c-9353-75d17347ad7d">


#### Test the "Smoke test release" workflow

1. Go to the [Smoke test release](https://github.com/woocommerce/woocommerce/actions/workflows/smoke-test-release.yml) page.
1. Select this branch `e2e/gutenberg-latest-nightly-0`
1. Enter any release version such as `7.0.0`.
1. Run the workflow.
1. Wait for all the jobs to finish.
1. Go to the [WooCommerce Test Reports Dashboard > Smoke Tests on Releases](https://woocommerce.github.io/woocommerce-test-reports/release/) page.
1. Verify that clicking the `E2E` links for `Gutenberg` and `Gutenberg - Nightly` will take you to the correct Allure reports
   - Make sure that the plugin name, time, and date shown in the Allure report are correct.
   <img width="351" alt="sW8EkXZRSw" src="https://github.com/woocommerce/woocommerce/assets/4509348/d079e846-d7c3-4039-99bd-59834474e570">
